external/vcm/vcm/cubedsphere/regridz.py

```diff
@@ -196,11 +225,53 @@ def _mask_weights(
             other=0.0,
         )
     else:
-        return weights.where(
-            phalf_coarse_on_fine.isel({dim_outer: slice(1, None)}).variable
-            < phalf_fine.isel({dim_outer: SURFACE_LEVEL}).variable,
-            other=0.0,
-        ).rename({dim_outer: dim_center})
+        strictly_interpolated = _compute_strictly_interpolated(
+            phalf_fine, phalf_coarse_on_fine, dim_center, dim_outer,
+        )
+        return weights.where(strictly_interpolated, other=0.0)
+
+
+def _compute_strictly_interpolated(
+    phalf_fine: xr.DataArray,
+    phalf_coarse_on_fine: xr.DataArray,
+    dim_center: str = RESTART_Z_CENTER,
+    dim_outer: str = RESTART_Z_OUTER,
+) -> xr.DataArray:
+    coarse_interfaces = phalf_coarse_on_fine.isel({dim_outer: slice(1, None)})
+    fine_surface_pressure = phalf_fine.isel({dim_outer: SURFACE_LEVEL})
+    result = coarse_interfaces < fine_surface_pressure
+    return result.rename({dim_outer: dim_center})
+
+
+def _adiabatically_adjust_extrapolated_temperature(
+    temperature: xr.DataArray,
+    pfull_fine: xr.DataArray,
+    pfull_coarse_on_fine: xr.DataArray,
+    phalf_fine: xr.DataArray,
+    phalf_coarse_on_fine: xr.DataArray,
+    z_dim_center: str = RESTART_Z_CENTER,
+    z_dim_outer: str = RESTART_Z_OUTER,
+) -> xr.DataArray:
+    strictly_interpolated = _compute_strictly_interpolated(
+        phalf_fine, phalf_coarse_on_fine, dim_center=z_dim_center, dim_outer=z_dim_outer
+    )
+
+    # Compute the potential temperature at all grid cells using the coarse model
+    # level midpoint pressures as a target and the fine lowest level midpoint
+    # pressure as a reference.
+    cp_air = 1004.6
```

Comment:
I would like to use: https://github.com/ai2cm/fv3net/blob/7874333ba36f5589cc43da66800c0a4f6d560727/external/vcm/vcm/calc/thermo/constants.py#L9
but it's oddly inconsistent with [what is used in FV3GFS](https://github.com/ai2cm/fv3gfs-fortran/blob/cb106f2eb806e8c635d28d8b76ee8e80a0e20bc3/FMS/constants/constants.F90#L55).  I may float the idea of changing it so that it is consistent with FV3GFS.

Reply:
Right, fv3gfs also uses a different set of constants depending on the compiler flag `GFS_PHYS`. Probably good to consolidate